repair/repair.cc

```diff
@@ -638,53 +638,61 @@ future<> shard_repair_task_impl::repair_range(const dht::token_range& range, ::t
     auto mandatory_neighbors = std::move(r_neighbors.mandatory);
     auto live_neighbors = boost::copy_range<std::vector<gms::inet_address>>(neighbors |
                 boost::adaptors::filtered([this] (const gms::inet_address& node) { return gossiper.is_alive(node); }));
-    for (auto& node : mandatory_neighbors) {
-        auto it = std::find(live_neighbors.begin(), live_neighbors.end(), node);
-        if (it == live_neighbors.end()) {
-            nr_failed_ranges++;
+
+    auto task_impl_ptr = std::make_unique<range_repair_task_impl>(rs.get_repair_module().shared_from_this(), tasks::task_id::create_random_id(), _status.keyspace, std::move(table_name), get_repair_uniq_id(), *this, range, table_id, std::move(neighbors), std::move(mandatory_neighbors), std::move(live_neighbors), ranges_index, _reason);
```

Comment:
Too many paramters. As a follow up, we can define a class for those parameters and pass a single object to range_repair_task_impl.